massa-protocol-worker/src/handlers/peer_handler/models.rs

```diff
@@ -72,8 +72,8 @@ impl PeerDB {
     }
 
     pub fn unban_peer(&mut self, peer_id: &PeerId) {
-        if self.peers.contains_key(peer_id) {
-            self.peers.remove(peer_id);
+        if let Some(peer) = self.peers.get_mut(peer_id) {
+            peer.state = PeerState::Trusted;
```

Comment:
don't know if the state should be Trusted.  
Trusted means the peer has been tested successfully.

Reply:
Changed to HandshakeFailed like you proposed.